### PR TITLE
Gauntlet - Aave V2 Ethereum AMM IR Deprecation

### PR DESCRIPTION
**Pull request by jbatch01**

### CODE REVIEW

**Comment by jbatch01** [comment]
We're running into two issues with this AIP:
- Running `make test-contract filter=AaveV2EthereumAMM_ARFCDeprecateAaveV2AMMMarketStep2_20240205` causes the following issue
```
    │   └─ ← ReserveData({ configuration: ReserveConfigurationMap({ data: 18446965362433828853616 [1.844e22] }), liquidityIndex: 1000000000000000000000000000 [1e27], variableBorrowIndex: 1000000000000000000000000000 [1e27], currentLiquidityRate: 0, currentVariableBorrowRate: 30000000000000000000000000 [3e25], currentStableBorrowRate: 0, lastUpdateTimestamp: 1679447915 [1.679e9], aTokenAddress: 0xCa5DFDABBfFD58cfD49A9f78Ca52eC8e0591a3C5, stableDebtTokenAddress: 0xFEaeCde9Eb0cd43FDE13427C6C7ef406780a8136, variableDebtTokenAddress: 0x0B7c7d9c5548A23D0455d1edeC541cc2AD955a9d, interestRateStrategyAddress: 0x52E39422cd86a12a13773D86af5FdBF5665989aD, id: 22 })
    ├─ [7834] 0xc443AD9DDE3cecfB9dfC5736578f447aFE3590ba::getReserveTokensAddresses(0xD2eeC91055F07fE24C9cCB25828ecfEFd4be0c41) [staticcall]
    │   ├─ [471] 0xAcc030EF66f9dFEAE9CbB0cd1B25654b82cFA8d5::getLendingPool() [staticcall]
    │   │   └─ ← 0x7937D4799803FbBe595ed57278Bc4cA21f3bFfCB
    │   ├─ [3718] 0x7937D4799803FbBe595ed57278Bc4cA21f3bFfCB::getReserveData(0xD2eeC91055F07fE24C9cCB25828ecfEFd4be0c41) [staticcall]
    │   │   ├─ [3047] 0xb9184a4480830bf89B55b73631E287Df9079F466::getReserveData(0xD2eeC91055F07fE24C9cCB25828ecfEFd4be0c41) [delegatecall]
    │   │   │   └─ ← ReserveData({ configuration: ReserveConfigurationMap({ data: 18446965362433828853616 [1.844e22] }), liquidityIndex: 1000000000000000000000000000 [1e27], variableBorrowIndex: 1000000000000000000000000000 [1e27], currentLiquidityRate: 0, currentVariableBorrowRate: 30000000000000000000000000 [3e25], currentStableBorrowRate: 0, lastUpdateTimestamp: 1679447915 [1.679e9], aTokenAddress: 0xCa5DFDABBfFD58cfD49A9f78Ca52eC8e0591a3C5, stableDebtTokenAddress: 0xFEaeCde9Eb0cd43FDE13427C6C7ef406780a8136, variableDebtTokenAddress: 0x0B7c7d9c5548A23D0455d1edeC541cc2AD955a9d, interestRateStrategyAddress: 0x52E39422cd86a12a13773D86af5FdBF5665989aD, id: 22 })
    │   │   └─ ← ReserveData({ configuration: ReserveConfigurationMap({ data: 18446965362433828853616 [1.844e22] }), liquidityIndex: 1000000000000000000000000000 [1e27], variableBorrowIndex: 1000000000000000000000000000 [1e27], currentLiquidityRate: 0, currentVariableBorrowRate: 30000000000000000000000000 [3e25], currentStableBorrowRate: 0, lastUpdateTimestamp: 1679447915 [1.679e9], aTokenAddress: 0xCa5DFDABBfFD58cfD49A9f78Ca52eC8e0591a3C5, stableDebtTokenAddress: 0xFEaeCde9Eb0cd43FDE13427C6C7ef406780a8136, variableDebtTokenAddress: 0x0B7c7d9c5548A23D0455d1edeC541cc2AD955a9d, interestRateStrategyAddress: 0x52E39422cd86a12a13773D86af5FdBF5665989aD, id: 22 })
    │   └─ ← 0xCa5DFDABBfFD58cfD49A9f78Ca52eC8e0591a3C5, 0xFEaeCde9Eb0cd43FDE13427C6C7ef406780a8136, 0x0B7c7d9c5548A23D0455d1edeC541cc2AD955a9d
    └─ ← revert: ERROR: No usable collateral found

Test result: FAILED. 0 passed; 1 failed; 0 skipped; finished in 1.21s
 
Ran 1 test suites: 0 tests passed, 1 failed, 0 skipped (1 total tests)
```

- The interest rate curve changes are not being reflected in the before/after .md file

File: src/20240205_AaveV2EthereumAMM_ARFCDeprecateAaveV2AMMMarketStep2/AaveV2EthereumAMM_ARFCDeprecateAaveV2AMMMarketStep2_20240205.t.sol [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  }
+
+  /**
+   * @dev executes the generic test suite including e2e and config snapshots
+   */
+  function test_defaultProposalExecution() public {
+    defaultTest(
+      'AaveV2EthereumAMM_ARFCDeprecateAaveV2AMMMarketStep2_20240205',
+      AaveV2EthereumAMM.POOL,
+      address(proposal)
```
Comment (sakulstra) [inline src/20240205_AaveV2EthereumAMM_ARFCDeprecateAaveV2AMMMarketStep2/AaveV2EthereumAMM_ARFCDeprecateAaveV2AMMMarketStep2_20240205.t.sol:29]:
```suggestion
      address(proposal),
      false
```
i think there's an overload to disable the e2e suite, while that's not great, rf updates have been done quite a few times so i guess it's acceptable. We'll work on improvements for the future.

**Comment by jbatch01** [comment]
@sakulstra - Thanks for the suggestion. Everything passes when I run the test command locally.

The last issue we're running into is that `diffs/AaveV2EthereumAMM_ARFCDeprecateAaveV2AMMMarketStep2_20240205_before_AaveV2EthereumAMM_ARFCDeprecateAaveV2AMMMarketStep2_20240205_after.md` does not reflect the interest rate parameter changes that we are also making in tandem with the rf changes

File: src/20240205_AaveV2EthereumAMM_ARFCDeprecateAaveV2AMMMarketStep2/AaveV2EthereumAMM_ARFCDeprecateAaveV2AMMMarketStep2_20240205.sol [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+import {IAaveV2ConfigEngine} from 'aave-helpers/v2-config-engine/IAaveV2ConfigEngine.sol';
+import {IV2RateStrategyFactory} from 'aave-helpers/v2-config-engine/IV2RateStrategyFactory.sol';
+
+/**
+ * @title [ARFC] Deprecate Aave V2 AMM Market - Step 2
+ * @author Gauntlet
+ * - Snapshot: https://snapshot.org/#/aave.eth/proposal/0x0ade316f52d5f763160ea15538a71a4682ae1b708864e8d33497d8de40ad9973
+ * - Discussion: https://governance.aave.com/t/arfc-deprecate-aave-v2-amm-market-step-2/16408/1
+ */
+contract AaveV2EthereumAMM_ARFCDeprecateAaveV2AMMMarketStep2_20240205 is AaveV2PayloadEthereum {
```
Comment (sakulstra) [inline src/20240205_AaveV2EthereumAMM_ARFCDeprecateAaveV2AMMMarketStep2/AaveV2EthereumAMM_ARFCDeprecateAaveV2AMMMarketStep2_20240205.sol:16]:
```suggestion
contract AaveV2EthereumAMM_ARFCDeprecateAaveV2AMMMarketStep2_20240205 is AaveV2PayloadEthereumAMM {
```

File: src/20240205_AaveV2EthereumAMM_ARFCDeprecateAaveV2AMMMarketStep2/AaveV2EthereumAMM_ARFCDeprecateAaveV2AMMMarketStep2_20240205.sol [hunk 3]
```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {AaveV2EthereumAMM, AaveV2EthereumAMMAssets, ILendingPoolConfigurator} from 'aave-address-book/AaveV2EthereumAMM.sol';
+import {AaveV2PayloadEthereum} from 'aave-helpers/v2-config-engine/AaveV2PayloadEthereum.sol';
```
Comment (sakulstra) [inline src/20240205_AaveV2EthereumAMM_ARFCDeprecateAaveV2AMMMarketStep2/AaveV2EthereumAMM_ARFCDeprecateAaveV2AMMMarketStep2_20240205.sol:5]:
```suggestion
import {AaveV2PayloadEthereumAMM} from 'aave-helpers/v2-config-engine/AaveV2PayloadEthereumAMM.sol';
```

**Pull request merged**